test/unit/Container.spec.ts

```diff
@@ -98,4 +113,14 @@ describe("Container", function() {
         Container.get(ExtraService).byeMessage.should.be.equal("buy world");
     });
 
+    it("should support class factories", function() {
```

Comment:
the problem with container instance (unlike singleton or static service) that we will need to tell  to DI implicitly what services are defined. Because decorators can't have instance of container where they can store data about decorated classes.